src/uu/cp/src/cp.rs

```diff
@@ -1078,114 +1100,132 @@ impl OverwriteMode {
     }
 }
 
+#[macro_export]
+macro_rules! handle_preserve {
+    ($preserve:expr, $copy_call:expr) => {
+        if $preserve.is_preserve() {
+            let result = $copy_call();
+            if $preserve == Preserve::Require {
+                result?;
+            }
+        }
+    };
+}
```

Comment:
This doesn't need to be a macro, a function would be enough:
```rust
fn handle_preserve(p: Preserve, f: fn() -> CopyResult<()>) -> CopyResult<()> {
     if p.is_preserve() {
          let result = f();
          if preserve == Preserve::Require {
              result?;
          }
     }
     Ok(())
}
```
Only difference is that you call it with a `?` at the end.

Reply:
good idea, don't like macros very much :P